keel/document/models.py

```diff
@@ -23,15 +23,18 @@ def add_attachments(self, files, user_id):
         if err_msg:
             raise DocumentInvalid(err_msg)
 
-        docs = []
-        for file in files.values():
-            docs.append(self.model(avatar =file, 
-                                    doc_type = Documents.GENERIC, 
-                                    owner_id = user_id,
-                                    original_name= file.name,
-                                    doc_pk= generate_unique_id(self.model.DOCUMENT_PREFIX)))
-
-        self.bulk_create(docs, batch_size=1000)
+        try:
+            docs = []
+            for file in files.values():
+                docs.append(self.model(avatar =file, 
+                                        doc_type = Documents.GENERIC, 
+                                        owner_id = user_id,
+                                        original_name= file.name,
+                                        doc_pk= generate_unique_id(self.model.DOCUMENT_PREFIX)))
+
+            self.bulk_create(docs, batch_size=1000)
+        except Exception as e:
+            raise DocumentInvalid(e)
```

Comment:
Please log the exception also...
Also we cant simply throw a DocumentInvalid error
what if the Db server was down while processing the bulk_create request....
or other different causes